### DISCUSSION TITLE
Dynamic connection factory based on incoming request

### DISCUSSION THREAD

**Discussion by david-malecek** [spring-projects/spring-amqp#2465] [Q&A] [answered]
Hi community 👋 

I would like to kindly ask you to help me to find my way around. I need to implement a _routing_ service which will send a rabbit-mq message using a particular `ConnectionFactory` based on incoming request type - rabbit-mq's `vhost` will be part of the request. Is the `RoutingConnectionFactory` the right way? Creating a connection factory on the fly for each request would work too but it would be rather inefficient I guess. What do you think, please?

Thank you and have a nice day!

**Comment by artembilan** [comment, marked as answer]
Well, since we don't know details about your task, it sounds really like a `RoutingConnectionFactory` is the way to go.
The `RestTemplate` can be configured with a `sendConnectionFactorySelectorExpression` where `org.springframework.amqp.core.Message` is used a root evaluation object.
See more in docs: https://docs.spring.io/spring-amqp/reference/html/#routing-connection-factory

**Reply by david-malecek** [reply]
Hi @artembilan , the `RoutingConnectionFactory` was the right way to go. Thank you for your help!

David
